core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -70,15 +70,17 @@ trait GraphQL[-R] { self =>
           schema.subscription.map(_.opType),
           additionalDirectives
         )
-      lazy val introspectionRootSchema: RootSchema[Any] = Introspector.introspect(rootType)
-      lazy val introspectionRootType: RootType          = RootType(introspectionRootSchema.query.opType, None, None)
+
+      lazy val introWrappers = wrappers.collect { case w: IntrospectionWrapper[R] => w }
```

Comment:
It doesn't seem like `Introspection` actually needs to be variant.

Reply:
won't compile otherwise because of 
`def introspect[R](rootType: RootType, introWrappers: List[IntrospectionWrapper[R]]): (RootSchema[Any], ZIO[R, ExecutionError, RootSchema[Any]])`